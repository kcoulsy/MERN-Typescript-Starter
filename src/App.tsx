import React from 'react';
import { connect } from 'react-redux';

import '../node_modules/fomantic-ui/dist/semantic.min.css';

import { initialise } from './actions/auth';

import AppRouter from './components/Router/AppRouter';
import NavBar from './components/NavBar/NavBar';

interface AppProps {
    initialise: Function;
}

export const App : React.FunctionComponent<AppProps> = ({ initialise }) => {
    initialise();

    return (
        <AppRouter>
            <div className="App">
                <NavBar />
                <h1>hello world</h1>
            </div>
        </AppRouter>
    );
};

const mapDispatchToProps = { initialise };

export default connect(undefined, mapDispatchToProps)(App);
