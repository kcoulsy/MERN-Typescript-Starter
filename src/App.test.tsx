import React from 'react';
import { shallow } from 'enzyme';

import { shallowToJson } from 'enzyme-to-json';
import { App } from './App';

describe('render AppRouter component', () => {
    it('should render AppRouter component', () => {
        const initialise = jest.fn();
        const wrapper = shallow(<App initialise={initialise} />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(initialise.mock.calls.length).toBeGreaterThan(0);
    });
});
