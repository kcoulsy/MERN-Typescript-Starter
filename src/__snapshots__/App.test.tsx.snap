// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AppRouter component should render AppRouter component 1`] = `
<AppRouter>
  <div
    className="App"
  >
    <Connect(Component) />
    <h1>
      hello world
    </h1>
  </div>
</AppRouter>
`;
